src/core/adapters/personalization.ts

```diff
@@ -0,0 +1,120 @@
+import Actions from '../actions';
+import Configuration from '../configuration';
+import Selectors from '../selectors';
+import Store from '../store';
+import ConfigAdapter from './configuration';
+
+namespace Personalization {
+  type ExtractableAction = Actions.SelectRefinement & Actions.AddRefinement;
+
+  export const DAYS_IN_SECONDS = 86400;
+
+  export const extractBias = (action: ExtractableAction, state: Store.State) => {
+    const config = Selectors.config(state).personalization.realTimeBiasing;
+    const byId = Selectors.realTimeBiasesById(state);
+    const { field, value } = extractRefinement(action, state);
+
+    if (config.attributes[field]) {
+      return {
+        variant: field,
+        key: value,
+        bias: (byId[field] && byId[field][value]) ? {
+          ...byId[field][value],
+          lastUsed: Math.floor(Date.now() / 1000)
+        } : generateNewBias()
+      };
+    }
+  };
+
+  // tslint:disable-next-line max-line-length
+  export const extractRefinement = ({ type, payload }: ExtractableAction, state: Store.State): { field: string, value: string } => {
+    switch (type) {
+      case Actions.ADD_REFINEMENT:
+        if (!payload.range) {
+          return { field: payload.navigationId, value: payload.value };
+        }
+        break;
+      case Actions.SELECT_REFINEMENT:
+        return Selectors.refinementCrumb(state, payload.navigationId, payload.index);
+    }
+
+    return { field: undefined, value: undefined };
+  };
+
+  export const generateNewBias = () => ({ lastUsed: Math.floor(Date.now() / 1000) });
+
+  // tslint:disable-next-line max-line-length
+  export const transformToBrowser = (state: Store.Personalization.Biasing, reducerKey: string): BrowserStorageState => ({
+    allIds: state.allIds.map(({ variant, key }) => ({
+      variant,
```

Comment:
I don't want to confuse the usage of the term `variant` which is specifically for a permutation of a product (perhaps a version of a product that comes in a given colour). The concept we want to use here is refinements which we represent in the rest of the system as `field` and `value` which I would like to use instead of `key` and `variant`